packages/web-pkg/src/components/QuotaSelect.vue

```diff
@@ -120,34 +108,46 @@ export default {
     optionSelectable(option) {
       return option.selectable !== false
     },
+    isValueValidNumber(value) {
+      const optionIsNumberRegex = /^[0-9]\d*(([.,])\d+)?$/g
+      return value !== '0' && optionIsNumberRegex.test(value)
+    },
     createOption(option) {
       option = option.replace(',', '.')
-      const optionIsNumberRegex = /^[0-9]\d*(([.,])\d+)?$/g
-      if (option === '0' || !optionIsNumberRegex.test(option)) {
+
+      if (!this.isValueValidNumber(option)) {
         return {
-          displayValue: option,
+          value: option,
           error: this.$gettext('Please enter only numbers'),
           selectable: false
         }
       }
       const value = parseFloat(option) * Math.pow(10, 9)
-      const displayValue = parseFloat(option).toFixed(2).replace('.00', '')
+
       if (this.maxQuota && value > this.maxQuota) {
         return {
           value,
-          displayValue,
-          displayUnit: 'GB',
-          error: this.$gettext('Please enter a value equal or less than %{ maxQuota } GB', {
-            maxQuota: this.maxQuota * Math.pow(10, -9)
+          error: this.$gettext('Please enter a value equal to or less than %{ maxQuota }', {
+            maxQuota: this.getFormattedFileSize(this.maxQuota)
+          }),
+
+          selectable: false
+        }
+      }
+
+      if (value > 1e15) {
```

Comment:
Can't you omit this case if you define the `maxQuota` with a fallback to `1e15` if it's undefined?

Reply:
Eh, it's a prop. So then just define a local variable in the scope of the `createOption` method:
```js
const maxQuota = this.maxQuota || 1e15
```

Reply:
GOOD CATCH THO!

Reply:
But we can overwrite the default value =)